src/datalinker/config.py

```diff
@@ -4,24 +4,51 @@
 
 from datetime import timedelta
 from pathlib import Path
-from typing import Annotated
+from typing import Annotated, Self
 
-from pydantic import Field, HttpUrl, SecretStr
+import yaml
+from pydantic import BaseModel, Field, HttpUrl, model_validator
+from pydantic.alias_generators import to_camel
 from pydantic_settings import BaseSettings, SettingsConfigDict
 from safir.logging import LogLevel, Profile
 from safir.pydantic import HumanTimedelta
 
-__all__ = [
-    "Config",
-    "config",
-]
+__all__ = ["Config", "HiPSDatasetConfig"]
+
+
+class HiPSDatasetConfig(BaseModel):
+    """Configuration for a single HiPS dataset."""
+
+    url: Annotated[
+        HttpUrl,
+        Field(title="Base URL", description="Base URL for this HiPS dataset"),
+    ]
+
+    paths: Annotated[
+        list[str],
+        Field(
+            title="HiPS paths",
+            description="List of available HiPS paths",
+            default_factory=lambda: [
+                "images/color_gri",
+                "images/color_riz",
+                "images/band_u",
+                "images/band_g",
+                "images/band_r",
+                "images/band_i",
+                "images/band_z",
+                "images/band_y",
+                "images/2MASS/Color",
+            ],
+        ),
+    ]
 
 
 class Config(BaseSettings):
     """Configuration for datalinker."""
 
     model_config = SettingsConfigDict(
-        env_prefix="DATALINKER_", case_sensitive=False
+        alias_generator=to_camel, extra="forbid", populate_by_name=True
```

Comment:
You should only need to set `alias_generator` *or* set `validation_alias` on each parameter that has underscores, not both. (I've been leaning towards doing the latter in new code because it feels a bit more explicit, but I think all approaches are easily defensible and it's a matter of personal taste.)